cmd/metal-api/internal/service/machine-service.go

```diff
@@ -269,6 +269,17 @@ func (r machineResource) webService() *restful.WebService {
 		Returns(http.StatusGatewayTimeout, "Timeout", httperrors.HTTPErrorResponse{}).
 		DefaultReturns("Error", httperrors.HTTPErrorResponse{}))
 
+	ws.Route(ws.POST("/{id}/reinstall").
+		To(editor(r.reinstallMachine)).
+		Operation("reinstallMachine").
+		Doc("reinstall this machine").
+		Param(ws.PathParameter("id", "identifier of the machine").DataType("string")).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(v1.MachineReinstallRequest{}).
+		Returns(http.StatusOK, "OK", v1.MachineResponse{}).
+		Returns(http.StatusGatewayTimeout, "Timeout", httperrors.HTTPErrorResponse{}).
```

Comment:
This endpoint does not return a status gateway timeout. I guess, you copied that from the wait endpoint. If this is the case, you can remove this line.